### PR TITLE
Add configuration to automatically freeze funds for Validators 

### PR DESCRIPTION
**Pull request by omerfirmak**
Faucet will now keep track of frozen funds of Validators, and create
one if they dont have any.

this will help us scale number of validators in our test network without
manually changing the genesis.

### CODE REVIEW

**Comment by hewison-chris** [comment]
What will trigger these validators to enroll?

**Comment by omerfirmak** [comment]
> What will trigger these validators to enroll?

recurring_enrollment config

**Review by hewison-chris — APPROVED** [review]
LGTM

File: source/faucet/main.d [hunk 1]
```diff
@@ -368,6 +375,12 @@ public class Faucet : FaucetAPI
         logInfo("\tL: %s, H: %s", sutxo[0].output.value, sutxo[$-1].output.value);
 
         logInfo("\tutxo owned by Faucet: %d entries", this.state.owned_utxos.length);
+
+        auto to_freeze_pks = validators.filter!((pk) {
+            return this.state.utxos.getUTXOs(pk).byValue.all!(utxo => utxo.output.type != OutputType.Freeze) &&
+                ((pk in this.state.freeze_txs) is null || !this.randomClient().hasTransactionHash(this.state.freeze_txs[pk].hashFull));
```
Comment (Geod24) [inline source/faucet/main.d:381]: `pk !in ...`

The `hasTransactionHash` is to prevent re-sending when we just restarted, right ?

**Review by Geod24 — APPROVED** [review]

**Comment by Geod24** [comment]
Merge and deploy at will

File: source/faucet/main.d (hunk 1 above)
Comment (omerfirmak) [reply source/faucet/main.d:381]: `hasTransactionHash` is there in case the TX we sent is no longer in the pool and we still have no frozen UTXO.

**Pull request merged**